processor/tailsamplingprocessor/sampling/status_code.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type statusCodeFilter struct {
+	logger      *zap.Logger
+	statusCodes []pdata.StatusCode
+}
+
+var _ PolicyEvaluator = (*statusCodeFilter)(nil)
+
+// NewStatusCodeFilter creates a policy evaluator that samples all traces with
+// a given status code.
+func NewStatusCodeFilter(logger *zap.Logger, statusCodeString []string) (PolicyEvaluator, error) {
+	if len(statusCodeString) == 0 {
+		return nil, errors.New("expected at least one status code to filter on")
+	}
+
+	statusCodes := make([]pdata.StatusCode, len(statusCodeString))
+
+	for i := range statusCodeString {
+		switch statusCodeString[i] {
+		case "OK":
+			statusCodes[i] = pdata.StatusCodeOk
+		case "ERROR":
+			statusCodes[i] = pdata.StatusCodeError
+		case "UNSET":
+			statusCodes[i] = pdata.StatusCodeUnset
+		default:
+			return nil, fmt.Errorf("unknown status code %s, supported: OK, ERROR, UNSET", statusCodeString)
```

Comment:
```suggestion
			return nil, fmt.Errorf("unknown status code %q, supported: OK, ERROR, UNSET", statusCodeString)
```

Reply:
Just noticed there was a bug in the error handling, fixed this now.